stats/DataAggregation.md

```diff
@@ -0,0 +1,67 @@
+# Data Aggregation API Overview
+The stats library allows users to define measurements how data are aggregated and broken down.
+The core data types used are:
+* `Aggregation`: defines the aggregation type (e.g. `Distribution`, `Sum`, `Mean`).
+* `View`: defines how individual measurements are broken down by tags and aggregated.
+
+## Aggregation
+An Aggregation describes how data collected is aggregated.
+
+The library SHOULD provide support for multiple types of Aggregations:
+* `Count`: indicates that data collected and aggregated with this `Aggregation` type will be turned
+into a count value.
+* `Sum`: indicates that data collected and aggregated with this `Aggregation` will be summed up.
+* `Mean`: indicates that data collected and aggregated with this `Aggregation` will calculate the
+mean value.
+* `Distribution`: indicates that the desired `Aggregation` is a histogram distribution. A
+distribution `Aggregation` may contain a histogram of the values in the population. User can
+define the bucket boundaries for that histogram.
+
+## View
+A View describes how individual measurements are both broken down (by a set of tag keys, which
+form the basis for a Metric labels.) and aggregated (e.g. into Distributions). A single Measure
+can be used by multiple Views.
+
+A View is defined from the following:
+* `name`: a string by which the View will be referred to, e.g. "rpc_latency". Names MUST be unique
```

Comment:
Should we mention the constraint on View names? Currently it should be a ASCII string with a length <= 255 characters.

Reply:
We can add later a section with restrictions for all types.